src/components/BottomNavigation.js

```diff
@@ -797,6 +808,25 @@ class BottomNavigation<T: *> extends React.Component<Props<T>, State> {
                             />
                           )}
                         </Animated.View>
+                        <View
+                          style={[
+                            styles.badgeContainer,
+                            {
+                              right:
+                                (badge != null && typeof badge !== 'boolean'
+                                  ? String(badge).length * -2
+                                  : 0) - 2,
+                            },
+                          ]}
+                        >
+                          {typeof badge === 'boolean' ? (
+                            <Badge visible={badge} size={8} />
+                          ) : (
+                            <Badge visible={badge != null} size={16}>
```

Comment:
Shouldn't it be !== ?
```suggestion
                            <Badge visible={badge !== null} size={16}>
```